### PR TITLE
Fix bare specifier rewrites for top-level-to-dependency imports

### PR DESCRIPTION
**Pull request by justinfagnani**
- [x] CHANGELOG.md has been updated

### CODE REVIEW

File: src/babel-plugin-bare-specifiers.ts [hunk 1]
```diff
@@ -29,61 +30,71 @@ type HasSpecifier =
...
                   packageJson['jsnext:main'] || packageJson.main;
               return packageJson;
             },
-          });
+      });
 
-          let relativeSpecifierUrl =
-              relative(dirname(filePath), resolvedSpecifier);
+      let relativeSpecifierUrl: string;
+      // If we have an import from that cross the top-level / componentDir
```
Comment (aomarks) [inline src/babel-plugin-bare-specifiers.ts]: from that

File: src/babel-plugin-bare-specifiers.ts [hunk 2]
```diff
@@ -29,61 +30,71 @@ type HasSpecifier =
...
+      // boundary, fix up the path to generate polyserve-style ../ cross-package
+      // imports.
+      if (isComponentRequest && componentDir !== undefined &&
+          rootDir !== undefined && resolvedSpecifier.startsWith(componentDir) &&
```
Comment (aomarks) [inline src/babel-plugin-bare-specifiers.ts]: I think startsWith isn't safe for this check unless we ensure there's a trailing slash

File: src/babel-plugin-bare-specifiers.ts (hunk 1 above)
Comment (justinfagnani) [reply src/babel-plugin-bare-specifiers.ts]: fixed

File: src/babel-plugin-bare-specifiers.ts (hunk 2 above)
Comment (justinfagnani) [reply src/babel-plugin-bare-specifiers.ts]: fixed

**Comment by justinfagnani** [comment]
PTAL

**Pull request merged**